core/capabilities/targets/write_target.go

```diff
@@ -108,20 +122,48 @@ func (cap *EvmWrite) Execute(ctx context.Context, request capabilities.Capabilit
 	if inputs.Report == nil {
 		// We received any empty report -- this means we should skip transmission.
 		cap.lggr.Debugw("Skipping empty report", "request", request)
-		callback := make(chan capabilities.CapabilityResponse)
-		go func() {
-			// TODO: cast tx.Error to Err (or Value to Value?)
-			callback <- capabilities.CapabilityResponse{
-				Value: nil,
-				Err:   nil,
-			}
-			close(callback)
-		}()
-		return callback, nil
+		return success(), nil
 	}
 
 	// TODO: validate encoded report is prefixed with workflowID and executionID that match the request meta
 
+	// Check whether value was already transmitted on chain
```

Comment:
We shouldn't import EVM into core. I was talking to Deividas about this since the Capability Registry Syncer has the same problem. Chain Reader should always get initialised through the Relayer, but right now the only way to do that is to create a PluginProvider which by default has a bunch of stuff that you don't need for the capability. We will explore an easier way to access Chain Reader for use cases like this in the next few days

Reply:
This is chain writer, won't it need to write to chain, or is the thought that we'll expose the relay for the interface and it'll use that too.  I figured this could just live in the relayer as a capability so it can have direct access to both?

Reply:
Its not in the relayer right now, if it does move there then its fine to directly access whatever the capability needs